libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -1334,22 +1342,20 @@ int HTTPClient::handleHeaderResponse()
     while(connected()) {
         size_t len = _client->available();
         if(len > 0) {
+            int headerSeparator = -1;
             String headerLine = _client->readStringUntil('\n');
 
             lastDataTime = millis();
 
             DEBUG_HTTPCLIENT("[HTTP-Client][handleHeaderResponse] RX: '%s'\n", headerLine.c_str());
 
             if (headerLine.startsWith(F("HTTP/1."))) {
-                if (_canReuse) {
-                    _canReuse = (headerLine[sizeof "HTTP/1." - 1] != '0');
-                }
+
+                _canReuse = _canReuse && (headerLine[sizeof "HTTP/1." - 1] != '0');
                 _returnCode = headerLine.substring(9, headerLine.indexOf(' ', 9)).toInt();
-                continue;
-            }
+                _canReuse = _canReuse && (_returnCode != '0');
```

Comment:
`_returnCode` is already parsed to an integer in line 1355, so don't use the character comparison

If you wanted to disallow connection reuse on a server error (playing it safe) I would suggest:
`_canReuse = _canReuse && (_returnCode > 0) && (_returnCode <500);`

Otherwise, the original review comment would be requested:
`_canReuse = _canReuse && (_returnCode != 0);`